File: rental_app.py
from item_service import ItemService
from rental_service import RentalService


class RentalApp:
    def __init__(self):
        self.item_service = ItemService()
        self.rental_service = RentalService(self.item_service)
        self.item_service.set_rental_service(self.rental_service)

    def run(self):
        self.item_service.load_sample_items()

        while True:
            self.print_menu()
            choice = input()
            if choice == "1":
                self.item_service.list_items()
            elif choice == "2":
                self.rental_service.rent_item(input)
            elif choice == "3":
                self.rental_service.list_active_rentals()
            elif choice == "4":
                print("Bye!")
                return
            else:
                print("Invalid option.")

    def print_menu(self):
        print("\nRental CLI:")
        print("1. See all items to be rented")
        print("2. Rent an item")
        print("3. See all active rentals")
        print("4. Exit")
        print("Choose an option: ", end="")

File: RentalApp.js
const readline = require("readline");
const ItemService = require("./ItemService");
const RentalService = require("./RentalService");

class RentalApp {
  constructor() {
    this.rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    });
    this.itemService = new ItemService();
    this.rentalService = new RentalService(this.itemService);
    this.itemService.setRentalService(this.rentalService);
  }

  run() {
    this.itemService.loadSampleItems();
    this.showMenu();
  }

  showMenu() {
    console.log("\nRental CLI:");
    console.log("1. See all items to be rented");
    console.log("2. Rent an item");
    console.log("3. See all active rentals");
    console.log("4. Exit");
    this.rl.question("Choose an option: ", choice => {
      switch (choice) {
        case "1":
          this.itemService.listItems();
          return this.showMenu();
        case "2":
          this.rentalService.rentItem(this.rl, () => this.showMenu());
          break;
        case "3":
          this.rentalService.listActiveRentals();
          return this.showMenu();
        case "4":
          console.log("Bye!");
          this.rl.close();
          break;
        default:
          console.log("Invalid option.");
          return this.showMenu();
      }
    });
  }
}

module.exports = RentalApp;
